### ISSUE TITLE
"with" + scripts has wrong scoping for unqualified variables

### ISSUE THREAD

**Issue by sorlok** [enigma-dev/enigma-dev#668]
If you create an object, and then use:

```
with (obj_whatever) {
    some_script();
}
```

...then some_script will incorrectly scope any variables that are not explicitly qualified with "self". For example:

```
//some_script() 
//This works:
self.myvar = 1;

//some_script() 
//This doesn't:
myvar = 1;
```

GM5 and GM:S both scope "myvar" properly. An example project is below; it should show "1" for all tests:
https://drive.google.com/file/d/0B1P7NepPcOslbFoxaVBOY2NNejA/edit?usp=sharing

(Note: Ignore the project's name; I originally thought this was an inheritance bug.)

If you look at IDE_EDIT_objectfunctionality.h, the following code is generated in each case:

```
//self.myvar = 1;
enigma::varaccess_myvar(int(self))= 1;

//myvar = 1;
with((self)) {
  enigma::varaccess_myvar(int(self))= 1;
}

//myvar = 1; OBJ_ version:
myvar = 1;

//Strangely, self.myvar = 1; does not get an OBJ_ version.
```

So, basically, the OBJ_ version is overriding the correct behavior of this script by using its own local copy of "myvar". In other words, the object that CALLS this script will have its version of "myvar" modified, so the enclosing scope of "with" is lost. 

I'll be digging into this further to see how to fix this, but would appreciate any suggestions.
